components/message.php

```diff
@@ -0,0 +1,504 @@
+		$thread_id = $args[0];
+
+		$user = $this->get_user_id_from_identifier( $assoc_args['user-id'] );
+		if ( ! $user ) {
+			WP_CLI::error( 'No user found by that username or ID.' );
+		}
+
+		WP_CLI::confirm( 'Are you sure you want to delete this thread(s) ?', $assoc_args );
+
+		parent::_delete( array( $thread_id ), $assoc_args, function( $thread_id ) {
+
+			if ( messages_delete_thread( $thread_id, $user->ID ) ) {
+				return array( 'success', 'Thread successfully deleted.' );
+			} else {
+				return array( 'error', 'Could not delete the thread.' );
+			}
+		} );
+	}
+
+	/**
+	 * Get a message.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <message-id>
+	 * : Identifier for the message.
+	 *
+	 * [--fields=<fields>]
+	 * : Limit the output to specific fields.
+	 * ---
+	 * default: All fields.
+	 * ---
+	 *
+	 * [--format=<format>]
+	 * : Render output in a particular format.
+	 * ---
+	 * default: table
+	 * options:
+	 *   - table
+	 *   - json
+	 *   - haml
+	 * ---
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     $ wp bp message get 5465
+	 *     $ wp bp message see 5454
+	 *
+	 * @alias see
+	 */
+	public function get( $args, $assoc_args ) {
+		$m = new BP_Messages_Message( $args[0] );
+		$message_arr = get_object_vars( $m );
+
+		if ( empty( $assoc_args['fields'] ) ) {
+			$assoc_args['fields'] = array_keys( $message_arr );
+		}
+
+		$formatter = $this->get_formatter( $assoc_args );
+		$formatter->display_item( $message_arr );
+	}
+
+	/**
+	 * Get a list of messages for a specific user.
+	 *
+	 * ## OPTIONS
+	 *
+	 * --user-id=<user>
+	 * : Identifier for the user. Accepts either a user_login or a numeric ID.
+	 *
+	 * [--<field>=<value>]
+	 * : One or more parameters to pass. See BP_Messages_Box_Template()
+	 *
+	 * [--fields=<fields>]
+	 * : Fields to display.
+	 *
+	 * [--count=<count>]
+	 * : How many messages to list.
+	 * ---
+	 * default: 10
+	 * ---
+	 *
+	 * [--format=<format>]
+	 * : Render output in a particular format.
+	 * ---
+	 * default: table
+	 * options:
+	 *   - table
+	 *   - ids
+	 *   - count
+	 *   - csv
+	 *   - json
+	 *   - haml
+	 * ---
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     $ wp bp message list --user-id=544 --format=count
+	 *     10
+	 *
+	 *     $ wp bp message list --user-id=user_login --count=3 --format=ids
+	 *     5454 45454 4545 465465
+	 *
+	 * @subcommand list
+	 */
+	public function _list( $_, $assoc_args ) {
+		$formatter = $this->get_formatter( $assoc_args );
+
+		$r = wp_parse_args( $assoc_args, array(
+			'box'          => 'sentbox',
+			'type'         => 'all',
+			'search'       => '',
+			'count'        => 10,
+		) );
+
+		$user = $this->get_user_id_from_identifier( $assoc_args['user-id'] );
+		if ( ! $user ) {
+			WP_CLI::error( 'No user found by that username or ID.' );
+		}
+
+		$type = ( ! in_array( $r['type'], $this->message_types(), true ) ) ? 'all' : $r['type'];
+		$box  = ( ! in_array( $r['box'], $this->message_boxes(), true ) ) ? 'sentbox' : $r['box'];
+
+		$inbox = new BP_Messages_Box_Template( array(
```

Comment:
We don't use template classes in other `list` subcommands, do we? I'm assuming you did it to save duplicating some code - `if ( 'notice' === $this->box )` etc - is that right? I don't see a huge problem doing it this way, I just want to understand the reasoning.

Reply:
@boonebgorges That's how I'm doing in the BP-REST, so I decided to replicate it here. No ultimate reasoning behind, just found easier to do with it.